utils/thunder.go

```diff
@@ -43,99 +57,148 @@ func GetThunderConfig(client *SSHClient) (*ThunderConfig, error) {
 
 // ConfigureThunderVirtualization sets up GPU virtualization for prototyping mode
 func ConfigureThunderVirtualization(client *SSHClient, instanceID, deviceID, gpuType string, gpuCount int, token, binaryHash string) error {
-	// Check if update is needed
-	existingConfig, _ := GetThunderConfig(client)
-	existingHash, _ := ExecuteSSHCommand(client, fmt.Sprintf("sha256sum %s 2>/dev/null | awk '{print $1}' || echo ''", thunderLibPath))
-	existingHash = strings.TrimSpace(existingHash)
-
-	needsUpdate := false
-	if existingConfig.DeviceID == "" || existingConfig.GPUType != gpuType || existingConfig.GPUCount != gpuCount {
-		needsUpdate = true
-	}
-	if existingHash != binaryHash {
-		needsUpdate = true
-	}
+	checkCmd := fmt.Sprintf(`cat %s 2>/dev/null || echo '{}'; echo '---SEPARATOR---'; sha256sum %s 2>/dev/null | awk '{print $1}' || echo ''`, thunderConfigPath, thunderLibPath)
+	checkOutput, _ := ExecuteSSHCommand(client, checkCmd)
 
-	if !needsUpdate {
-		return nil
+	var existingConfig ThunderConfig
+	var existingHash string
+	parts := strings.SplitN(checkOutput, "---SEPARATOR---", 2)
+	if len(parts) >= 1 {
+		_ = json.Unmarshal([]byte(strings.TrimSpace(parts[0])), &existingConfig)
 	}
-
-	// Create Thunder directory
-	if _, err := ExecuteSSHCommand(client, fmt.Sprintf("mkdir -p %s", thunderConfigDir)); err != nil {
-		return fmt.Errorf("failed to create Thunder directory: %w", err)
+	if len(parts) >= 2 {
+		existingHash = strings.TrimSpace(parts[1])
 	}
 
-	// Download Thunder binary
-	downloadCmd := fmt.Sprintf("curl -L %s -o /tmp/libthunder.tmp && mv /tmp/libthunder.tmp %s", thunderBinaryURL, thunderLibPath)
-	if _, err := ExecuteSSHCommand(client, downloadCmd); err != nil {
-		return fmt.Errorf("failed to download Thunder binary: %w", err)
-	}
+	configNeedsUpdate := existingConfig.DeviceID == "" || existingConfig.GPUType != gpuType || existingConfig.GPUCount != gpuCount
 
-	// Create symlink
-	symlinkCmd := fmt.Sprintf("sudo mkdir -p /etc/thunder && sudo ln -sf %s %s", thunderLibPath, thunderSymlink)
-	if _, err := ExecuteSSHCommand(client, symlinkCmd); err != nil {
-		return fmt.Errorf("failed to create Thunder symlink: %w", err)
+	binaryNeedsUpdate := false
+	if binaryHash != "" && existingHash != binaryHash {
```

Comment:
If binaryHash == "" (which I believe means we failed to get the hash from the bucket), we should assume the hash changed and we should update the binary